include/clap/ext/draft/triggers.h

```diff
@@ -0,0 +1,132 @@
+#pragma once
+
+#include "../../plugin.h"
+#include "../string-sizes.h"
+
+static CLAP_CONSTEXPR const char CLAP_EXT_TRIGGERS[] = "clap.triggers.draft/0";
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+/// @page Trigger events
+///
+/// This extension enables the plugin to provide a set of triggers to the host.
+///
+/// Some examples for triggers:
+/// - start recording into a plugin-internal audio/note buffer
+/// - trigger a retrospective audio/note looper
+/// - trigger a sample-and-hold unit (maybe even per-voice)
+/// - retrigger an envelope during voice lifetime
+/// - shake event for a spring-reverb
+
+enum {
+   // When set:
+   // - trigger events can be recorded
+   // - trigger events can be played back
+   //
+   // The host can send live user trigger events regardless of this flag.
+   //
+   // If this trigger affects the internal processing structure of the plugin, ie: max delay, fft
+   // size, ... and the plugins needs to re-allocate its working buffers, then it should call
+   // host->request_restart(), and perform the change once the plugin is re-activated.
+   CLAP_TRIGGER_IS_AUTOMATABLE = 1 << 0,
+
+   // Does this trigger support per note automations?
+   CLAP_TRIGGER_IS_AUTOMATABLE_PER_NOTE_ID = 1 << 1,
+
+   // Does this trigger support per key automations?
+   CLAP_TRIGGER_IS_AUTOMATABLE_PER_KEY = 1 << 2,
+
+   // Does this trigger support per channel automations?
+   CLAP_TRIGGER_IS_AUTOMATABLE_PER_CHANNEL = 1 << 3,
+
+   // Does this trigger support per port automations?
+   CLAP_TRIGGER_IS_AUTOMATABLE_PER_PORT = 1 << 4,
+
+   // This trigger will affect the plugin output and requires to be done via
+   // process() if the plugin is active.
+   //
+   // A simple example would be a sample-and-hold, triggering it will change the output signal and must be
+   // processed.
+   CLAP_TRIGGER_REQUIRES_PROCESS = 1 << 5,

```

Comment:
Because the trigger is stateless, it isn't something that you can delay and send later on.
I think that the trigger must require process.
It is like a note event, you can't delay those, and they force processing.